Autofmt

bolliger32 opened the pull request:
- [x] tests added / passed (N/A)
 - [x] docs reflect changes (N/A)
 - [x] passes ``flake8 rhg_compute_tools tests docs`` (some things in tests still raise flake8 warnings, but they are ones that are needed for pytest)
 - [x] entry in HISTORY.rst (N/A

Started off by just calling black and flake8 on everything. Flake8 caused me to spot a few typos that hadn't caught up to us yet but would soon.

brews commented:
Yo, @bolliger32! I'm happy to give this a review this week. Could you give me a quick summary of what's going on and the goals here, I want to be sure I'm on the same page.

So we're hitting all the files with `black` and `flake8` and we're getting the cleaned up files here. What rules/configs were used for the `black` and `flake8`? 

What is the thinking about making `black` and `flake8` checks part of an early test stage in CI/CD? i.e.: CI/CD rejects code changes unless it follows style?

bolliger32 commented:
Hey @brews ! 

Yeah the summary is just that I started off just trying to clean things up just calling black on everything. We've added `black --check .` as part of our testing on the coastal team repo and I think it's helped keep things fairly uniform. So i'd be in favor of that if you guys want. Regardless, for the time being, I just thought I would tidy things up a bit. In addition to black, I also ran flake8 just to see what it would come up with (just default settings, except for --max-line-length=88 to match black). I didn't fix everything I dont' think (e.g. I didn't fix all whitespaces) but I fixed the typos it noticed (like when I wrote isintance instead of isinstance).

Previously master was failing so I thought that was the failures that this PR was running into, but I just rebased onto @delgadom 's changes (which are now passing on master) and it's still failing. Not sure what's going on as it's a sphinx-build issue that passes fine on my notebook...

delgadom commented on .travis.yml:
```diff
@@ -53,7 +53,7 @@ script:
   then
     pytest &&
-    rm -f docs/rhg_compute_tools.*.rst &&
+    rm -f docs/rhg_compute_tools*.rst docs/modules.rst &&
```
Actually excluding docs/modules is intentional. This allows us to have our docs section be called "API" instead of "rhg_compute_tools" which is confusing. If you could roll back this change and the changes to `docs/modules.rst` ([here](https://github.com/RhodiumGroup/rhg_compute_tools/pull/71/files#diff-048b1487c1c99241f36bcfebe3679f05R1)) that would be great!

bolliger32 replied:
done

delgadom commented on docs/conf.py:
```diff
@@ -13,14 +13,16 @@
 import os
+import sys
+
+import rhg_compute_tools
```
This doesn't work. Sphinx is run from the docs directory, so the `sys.path.insert` below is required before `import rhg_compute_tools`. It's not strict PEP8, but that's probably what's causing your problem.

delgadom replied:
it's probably working on your system because you have rhg_compute_tools already installed I'm guessing?

bolliger32 replied:
that sounds correct to me! fixed.

bolliger32 commented:
> In general reviewing "black everything" PRs is close to impossible so I'd strongly prefer separating this from everything else. Ok if we close this, run black as a separate PR, then try to tackle these remaining items one at a time?

Soundsd good to me! I'll keep this branch and then when we're ready to try again after theres a separate black-only PR we can merge again.
